components/threading/channel_threads/thread_footer/thread_footer.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+/* eslint-disable react/prop-types */
+
+import React, {memo, ComponentProps} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import './thread_footer.scss';
+
+import Avatars from 'components/widgets/users/avatars';
+
+import Button from 'components/threading/common/button';
+import FollowButton from 'components/threading/common/follow_button';
+
+import Timestamp from 'components/timestamp';
+import SimpleTooltip from 'components/simple_tooltip';
+import ReplyIcon from 'components/widgets/icons/reply_icon';
+
+import {THREADING_TIME} from '../../common/options';
+
+type Props = {
+    participants: ComponentProps<typeof Avatars>['users'];
+    totalReplies: number;
+    newReplies: number;
+    lastReplyAt: ComponentProps<typeof Timestamp>['value'];
+    isFollowing: boolean;
+    actions: {
+        follow: () => void,
+        unfollow: () => void,
+        requestOpenThread: () => void;
+    }
+};
+
+function ThreadFooter({
+    participants,
+    totalReplies = 0,
+    newReplies = 0,
+    lastReplyAt,
+    isFollowing,
+    actions: {
+        follow,
+        unfollow,
+        requestOpenThread: open,
+    },
+}: Props) {
+    return (
+        <div className='ThreadFooter'>
+            {Boolean(newReplies) && <>
+                <SimpleTooltip
+                    id='threadFooterIndicator'
+                    content={
+                        <FormattedMessage
+                            id='threading.numNewMessages'
+                            defaultMessage='{newReplies, plural, =0 {no unread messages} =1 {one unread message} other {# unread messages}}'
+                            values={{newReplies}}
+                        />
+                    }
+                >
+                    <div className='indicator'>
+                        <div className='dot-unreads'/>
+                    </div>
+                </SimpleTooltip>
+            </>}
+
+            <Avatars
+                users={participants}
+                size='sm'
+            />
+
+            <Button
+                onClick={() => open()}
```

Comment:
This is creating a new function every render, can we just pass here `open`? as far as open is not handling any parameter, should be ok.

Reply:
We could, but I usually like to keep event handlers and actions/functionality separate to make it easier to handle interactions a little easier and in this case prevent future type collisions if the actions ever get parameters.

Open to thought. Should I switch it to a class component and put the event handlers there?

Reply:
as a functional component there is no much to do, probably creating the function and store it in a module level variable, but create it in the context of the function, something like:

```js
var open = null;
function Component() {
  open = open || () => {
    this.props.actions.whatever()
  }
}
```

Reply:
✓ Dropped the inline functions, and where necessary used `useCallback`